File: java/src/main/java/org/app/infrastructure/endpoints/dto/GetUserDTO.java
package org.app.infrastructure.endpoints.dto;

import org.app.domain.User;

public class GetUserDTO {
    public String id;
    public String username;
    public Integer followings;
    public Integer friends;
    public Integer fans;

    public GetUserDTO() {}

    public GetUserDTO(User user) {
        this.id = user.getId().toString();
        this.username = user.getUsername();
        this.followings = user.getFollowings().size();
        this.friends = user.getFriends().size();
        this.fans = user.getFans().size();
    }

    public static GetUserDTO from(User user){
        return new GetUserDTO(user);
    }
}

File: go/app/main/infrastructure/endpoints/dto/get_user_dto.go
package dto

import "mutualAttentionSystem/app/main/domain"

type GetUserDTO struct {
    Id         string `json:"id"`
    Username   string `json:"username"`
    Followings int    `json:"followings"`
    Friends    int    `json:"friends"`
    Fans       int    `json:"fans"`
}

func ToGetUserDTO(user *domain.User) *GetUserDTO {
    return &GetUserDTO{user.ID.String(), user.Username, len(user.GetFollowings()),
        len(user.GetFriends()), len(user.GetFans())}
}
